docs/illixr_plugins.md

```diff
@@ -1,85 +1,193 @@
+    Implements the `pose_prediction` service (defined in `common`),
+        so poses can be served directly to other plugins.
+
+    Topic details:
+
+    -   Asynchronously *reads* `pose_type` on `slow_pose` topic,
+            but it is only used as a fallback.
+    -   Asynchronously *reads* `imu_raw` on `imu_raw` topic.
+    -   Asynchronously *reads* `pose_type` on `true_pose` topic,
+            but it is only used if the client asks for the true pose.
+    -   Asynchronously *reads* `time_type` on `vsync_estimate` topic.
+        This tells `pose_predict` what time to estimate for.
 
 -   [`gldemo`][5]:
-    Renders a static scene (into left and right [_eye buffers_][34]) given the pose
+    Renders a static scene (into left and right [_eye buffers_][34]) given the [_pose_][37]
         from `pose_prediction`.
 
+    Topic details:
+
+    -   *Publishes* `rendered_frame` on `eyebuffer` topic.
+    -   Asynchronously *reads* `time_type` on `vsync_estimate` topic.
+
 -   [`timewarp_gl`][6]:
-    [Asynchronous reprojection][34] of the [_eye buffers_][34].
+    [Asynchronous reprojection][35] of the [_eye buffers_][34].
+    The timewarp ends just after [_vsync_][34], so it can deduce when the next vsync will be.
 
--   [`debugview`][7]: Renders a frame for debug information.
+    Topic details:
 
--   [`audio`][8]:
-    ***TODO.***
-    By default this is disabled, since it requires an NVIDIA GPU.
+    -   *Calls* `pose_prediction`.
+    -   Asynchronously *reads* `rendered_frame` on `eyebuffer` topic.
+    -   *Publishes* `time_type` on `mtp` topic, which is an estimate of the _nominal_ motion-to-photon latency.
+    -   *Publishes* `time_type` on `warp_frame_age` topic.
+    -   *Publishes* `type_type` on `vsync_estimate`.
+    -   *Publishes* `hologram_input` on `hologram_in` topic.
+    -   *Publishes* `texture_pose` on `texture_pose` topic if `ILLIXR_OFFLOAD_ENABLE` is set in the env.
 
--   [`hologram`][9]:
-    ***TODO.***
-    By default this is disabled, since it requires an NVIDIA GPU.
+-   [`debugview`][7]:
+    Renders incoming [_frames_][34] from the graphics pipeline for debugging live executions of the application.
+
+    Topic details:
+
+    -   *Calls* `pose_prediction`.
+    -   Asynchronously *reads* `imu_raw` on `imu_raw` topic.
+
+-   [`audio_pipeline`][8]:
+    Launches a thread for [binaural][19] recording and one for binaural playback.
+    Audio output is not yet routed to the system's speakers or microphone,
+        but the plugin's compute workload is still representative of a real system.
+    By default this plugin is disabled.
+
+    Topic details:
+
+    -   *Calls* `pose_prediction`.
 
 Below this point, we will use Switchboard terminology.
 Read the [API documentation on _Switchboard_][32] for more information.
 
 <img
-    src="../dataflow.dot.png"
-    alt ="ILLIXR dataflow graph generated by dataflow.dot"
+    src="../dataflow.png"
```

Comment:
Yup the name should remain `dataflow.dot.png`. I think the images are rendered assuming that they are in `docs/images`, so I will move the `.png` there so that we don't need the `..`.

Reply:
> Yup the name should remain `dataflow.dot.png`. I think the images are rendered assuming that they are in `docs/images`, so I will move the `.png` there so that we don't need the `..`.

This doesn't quite work. The images on a page are assumed to be in a directory with the same name as the markdown. After moving to `docs/images/dataflow.dot.png`, we need to specify the path as `../images/dataflow.dot.png`.